x-pack/plugins/ml/public/components/job_selector/job_selector.js

```diff
@@ -0,0 +1,510 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+
+import React, { useState, useEffect, useRef } from 'react';
+import { PropTypes } from 'prop-types';
+import moment from 'moment';
+
+import { ml } from '../../services/ml_api_service';
+import { JobSelectorTable } from './job_selector_table/';
+import { timefilter } from 'ui/timefilter';
+import { tabColor } from '../../../common/util/group_color_utils';
+import { getGroupsFromJobs, normalizeTimes, setGlobalState } from './job_select_service_utils';
+import { toastNotifications } from 'ui/notify';
+import {
+  EuiBadge,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiLink,
+  EuiSwitch,
+  EuiText,
+  EuiTitle
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+
+export function getBadge({ id, icon, isGroup = false, removeId, numJobs }) {
+  const color = isGroup ? tabColor(id) : 'hollow';
+  let props = { color };
+  let jobCount;
+
+  if (icon === true) {
+    props = {
+      ...props,
+      iconType: 'cross',
+      iconSide: 'right',
+      onClick: () => removeId(id),
+      onClickAriaLabel: 'Remove id'
+    };
+  }
+
+  if (numJobs !== undefined) {
+    jobCount = i18n.translate('xpack.ml.jobSelector.selectedGroupJobs', {
+      defaultMessage: `({jobsCount, plural, one {# job} other {# jobs}})`,
+      values: { jobsCount: numJobs },
+    });
+  }
+
+  return (
+    <EuiBadge key={`${id}-id`} {...props} >
+      {`${id}${jobCount ? jobCount : ''}`}
+    </EuiBadge>
+  );
+}
+
+function mergeSelection(jobIds, groupObjs, singleSelection) {
+  if (singleSelection) {
+    return jobIds;
+  }
+
+  const selectedIds = [];
+  const alreadySelected = [];
+
+  groupObjs.forEach((group) => {
+    selectedIds.push(group.groupId);
+    alreadySelected.push(...group.jobIds);
+  });
+
+  jobIds.forEach((jobId) => {
+    // Add jobId if not already included in group selection
+    if (alreadySelected.includes(jobId) === false) {
+      selectedIds.push(jobId);
+    }
+  });
+
+  return selectedIds;
+}
+
+function getInitialGroupsMap(selectedGroups) {
+  const map = {};
+
+  if (selectedGroups.length) {
+    selectedGroups.forEach((group) => {
+      map[group.groupId] = group.jobIds;
+    });
+  }
+
+  return map;
+}
+
+const BADGE_LIMIT = 10;
+const DEFAULT_GANTT_BAR_WIDTH = 299; // pixels
```

Comment:
Would there be a way to make the gantt bars work with percentages? Just an idea, change `DEFAULT_GANTT_BAR_WIDTH` to `100` and do the styles in `<TimeRangeBar />` with`%` instead of `px`. Might that work or are there some possible pitfalls (e.g. with flexbox) I'm missing?